src/Group.php

```diff
@@ -31,6 +31,25 @@ public function __construct(?string $prefix = null, ?callable $callback = null,
         return $factory($prefix, $routes);
     }
 
+    final public function setContainer(ContainerInterface $container): Group
```

Comment:
Since it's immutable, naming it with `set` isn't a good idea. Also, `self` should be used as return type since cloned object may be a super-class of `Group`.

```suggestion
    final public function withContainer(ContainerInterface $container): self
```

Reply:
Ok